### ISSUE TITLE
Json deserialization issue

### ISSUE THREAD

**Issue by nemo83** [doriordan/skuber#175]
Trying to `get[Endpoints]` and got this one:

```2018-07-18 11:22:46,391 [application-akka.actor.default-dispatcher-27] INFO skuber.api - [ { reqId=f3c064f2-3a3a-488e-bf0b-72d548fe2bae} } - about to send HTTP request: GET https://kubernetes.default/api/v1/namespaces/default/endpoints/api-neighbors] 
2018-07-18 11:22:46,809 [application-akka.actor.default-dispatcher-11] INFO skuber.api - [ { reqId=f3c064f2-3a3a-488e-bf0b-72d548fe2bae} } - received response with HTTP status 200] 
e => JsError(List((/subsets(0)/ports(0)/name,List(JsonValidationError(List(error.path.missing),WrappedArray()))), (/subsets(0)/notReadyAddresses,List(JsonValidationError(List(error.path.missing),WrappedArray())))))```

With curl I've replicated the same api call and effectively these fields are missing. 

```{
  "kind": "Endpoints",
  "apiVersion": "v1",
  "metadata": {
    "name": "api-neighbors",
    "namespace": "default",
    "selfLink": "/api/v1/namespaces/default/endpoints/api-neighbors",
    "uid": "c7a3c6a7-83bd-11e8-afbd-0e54929573ac",
    "resourceVersion": "3349824",
    "creationTimestamp": "2018-07-09T21:19:35Z"
  },
  "subsets": [
    {
      "addresses": [
        {
          "ip": "192.168.194.228",
          "nodeName": "ip-192-168-200-182.ec2.internal",
          "targetRef": {
            "kind": "Pod",
            "namespace": "default",
            "name": "api-neighbors-5f84458df5-68pws",
            "uid": "cdabd808-8a7b-11e8-afbd-0e54929573ac",
            "resourceVersion": "3349822"
          }
        },
        {
          "ip": "192.168.83.120",
          "nodeName": "ip-xxx-xxx-xxx-xxx.ec2.internal",
          "targetRef": {
            "kind": "Pod",
            "namespace": "default",
            "name": "api-neighbors-5f84458df5-x68px",
            "uid": "37bb80ea-8a65-11e8-afbd-0e54929573ac",
            "resourceVersion": "3335191"
          }
        }
      ],
      "ports": [
        {
          "port": 9000,
          "protocol": "TCP"
        }
      ]
    }
  ]
}```

**Comment by ngortheone** [comment]
@nemo83 can this issue be closed?

**Comment by nemo83** [comment]
That should be fixed. I'm waiting on a release so that I can then test and
close accordingly

thanks

Il giorno lun 23 lug 2018 alle ore 19:58 Ihor Antonov <
notifications@github.com> ha scritto:

**Comment by doriordan** [comment]
v2.0.9 released today has the fix
https://github.com/doriordan/skuber/releases/tag/v2.0.9

**Comment by doriordan** [comment]
closing this as should be fixed by 2.0.9, please open another issue if not the case
